File: python3/easy/power_of_thor.py
light_x, light_y, initial_tx, initial_ty = [int(i) for i in input().split()]
current_x = initial_tx
current_y = initial_ty

while True:
    remaining_turns = int(input())
    result = ""

    if current_y < light_y:
        result += "S"
        current_y += 1
    elif current_y > light_y:
        result += "N"
        current_y -= 1

    if current_x > light_x:
        result += "W"
        current_x -= 1
    elif current_x < light_x:
        result += "E"
        current_x += 1

    print(result)

File: java/pl/rozart/codingame/easy/PowerOfThor.java
package pl.rozart.codingame.easy;

import java.util.Scanner;

public class PowerOfThor {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int lightX = in.nextInt();
        int lightY = in.nextInt();
        int initialTX = in.nextInt();
        int initialTY = in.nextInt();

        int currentPosX = initialTX;
        int currentPosY = initialTY;
        while (true) {
            int remainingTurns = in.nextInt();

            String move = "";

            if (lightY > currentPosY) {
                move += "S";
                currentPosY++;
            } else if (lightY < currentPosY) {
                move += "N";
                currentPosY--;
            }

            if (lightX > currentPosX) {
                move += "E";
                currentPosX++;
            } else if (lightX < currentPosX) {
                move += "W";
                currentPosX--;
            }
            System.out.println(move);
        }
    }
}